melview.py

```diff
@@ -119,15 +119,24 @@ def get_cookie(self):
 
 # ---------------------------------------------------------------
 
+class MelViewZone:
+    def __init__(self, id, name, status):
+        self.id = id
+        self.name = name
+        self.status = status
+
+# ---------------------------------------------------------------
+
 class MelViewDevice:
     """ Handler class for a melview unit.
     """
 
-    def __init__(self, deviceid, buildingid, friendlyname,
+    def __init__(self, deviceid, buildingid, friendlyname, zones,
                  authentication, localcontrol=False):
         self._deviceid = deviceid
         self._buildingid = buildingid
         self._friendlyname = friendlyname
+        self._zones = zones
```

Comment:
Can you move this below, under `self._otemp_list` and just set it to an empty dictionary - it'll get updated almost right away in the call to `self._refresh_device_info()`.

Reply:
You'll need to remove the `zones` parameter above too.